custom_components/variable/helpers.py

```
@@ -120,162 +132,115 @@ def to_num(s):
-    else:
-        _LOGGER.debug(f"Invalid initial type: {type(init_val)}, returning None")
-        raise ValueError(f"Invalid initial type: {type(init_val)}")
-        return None
+        return _string_to_type(init_val, dest_type)
+    if isinstance(init_val, (int, float)):
+        return _number_to_type(init_val, dest_type)
+    if type(init_val) is datetime.date:
+        return _date_to_type(init_val, dest_type)
+    if type(init_val) is datetime.datetime:
+        return _datetime_to_type(init_val, dest_type)
+    raise ValueError(f"Invalid initial type: {type(init_val)}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)